### PR TITLE
Fix N+1 on variants when loading incoming exchange products

### PR DESCRIPTION
**Pull request by sauloperez**
#### What? Why?

Fixes the following N+1

```
user: root
GET /api/exchanges/1/products.json
USE eager loading detected
  Spree::Product => [:variants]
  Add to your query: .includes([:variants])
Call stack
  /usr/src/app/app/serializers/api/admin/for_order_cycle/supplied_product_serializer.rb:28:in `variants'
  /usr/src/app/app/controllers/api/exchange_products_controller.rb:89:in `render_paginated_products'
  /usr/src/app/app/controllers/api/exchange_products_controller.rb:29:in `index'
  bin/rails:4:in `require'
  bin/rails:4:in `<main>'
```

#### What should we test?

Green build.

#### Release notes
Fix N+1 on variants when loading incoming exchange products
Changelog Category: Technical changes

### CODE REVIEW

File: app/serializers/api/admin/for_order_cycle/supplied_product_serializer.rb [hunk 1]
```diff
@@ -19,13 +19,7 @@ def master_id
         end
 
         def variants
-          variants = if order_cycle.present? &&
-                        order_cycle.prefers_product_selection_from_coordinator_inventory_only?
```
Comment (Matt-Yorkley) [inline app/serializers/api/admin/for_order_cycle/supplied_product_serializer.rb:23]: This conditional is deleted without replacement. Is that logic not needed? Could you explain it a bit?

Comment (sauloperez) [reply app/serializers/api/admin/for_order_cycle/supplied_product_serializer.rb:23]: it is. If you go commit by commit you'll see. However, it was a rough attempt and I need to see if this is enough or not.

**Comment by mkllnk** [comment]
> this is still a rough attempt and I need to see if this is enough or not.

Should this be in Code Review?

**Comment by sauloperez** [comment]
This is still valid and an issue but I got no room for it now.

For the next dev: essentially conditional and activerecord login in a serializer is a code smell and should raise the alarms. The code in a serializer is executed in the context of an entry in an array, so whatever Activerecord Relation gets executed at that point, it'll cause an N+1.

Said queries should rather be pulled out and merged with the Relation that is built in the controller action so that we get all the data we need from the DB in a single shot.

**Pull request closed without merging**